dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3535,6 +3544,53 @@ prettyTypeMessage (MissingField k expr0) = ErrorMessages {..}
         txt0 = insert k
         txt1 = insert expr0
 
+prettyTypeMessage (ProjectionTypeMismatch k expr0 expr1) = ErrorMessages {..}
+  where
+    short = "Projection type mismatch\n"
+        <>  "\n"
+        <>  prettyDiff expr0 expr1
```

Comment:
This will only show the difference between the field types, right?!

In #1022 I was diffing the selector and its intersection with the record type which means that the field label is included…